backend/director/tools/kling.py

```
@@ -166,20 +193,129 @@ def text_to_video(self, prompt: str, save_at: str, duration: float, config: dict
+                    video_url = response.json()["data"]["task_result"]["videos"][0]["url"]
+                    # Download and save the video
+                    video_response = requests.get(video_url)
+                    video_response.raise_for_status()
+                    with open(save_at, "wb") as f:
+                        f.write(video_response.content)
+                    return
                 time.sleep(self.polling_interval)
-                continue
+
+        except Exception as e:
+            raise Exception(f"Error in KlingAITool.image_to_video: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)